test/units/testsuite-19.cleanup-slice.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+set -eux
+set -o pipefail
+
+# shellcheck source=test/units/util.sh
+. "$(dirname "$0")"/util.sh
+
+export SYSTEMD_LOG_LEVEL=debug
+
+# Create service with KillMode=none inside a slice
+cat <<EOF >/run/systemd/system/test19cleanup.service
+[Unit]
+Description=Test 19 cleanup Service
+[Service]
+Slice=test19cleanup.slice
+Type=simple
+ExecStart=sleep infinity
+KillMode=none
+EOF
+cat <<EOF >/run/systemd/system/test19cleanup.slice
+[Unit]
+Description=Test 19 cleanup Slice
+EOF
+
+# Start service
+systemctl start test19cleanup.service
+assert_rc 0 systemd-cgls /test19cleanup.slice
+
+# Stop slice
+# The sleep process will not be killed because of KillMode=none
+# Since there is still a process running under it, the /test19cleanup.slice cgroup won't be removed
+systemctl stop test19cleanup.slice
+
+# Kill sleep process manually
+while pkill sleep
+do
```

Comment:
`while …; do` please, to match what the usual style.

This doesn't really solve the race. We will wait at least 0.1 s after the process disappears, but the cgroup cleanup could take longer than that. Let's see if the tests pass, but even if they do, this seems very fragile. (Unfortunately the CI often runs in very overloaded systems, so any "sane" timeout is not enough. We really need either race-free things or retries or very long timeouts.)

Reply:
Hmm, but looking at this:
```
19:11:15 [    8.666079] testsuite-19.sh[450]: + pkill sleep
19:11:15 [    8.680801] testsuite-19.sh[450]: + assert_rc 1 systemd-cgls /test19cleanup.slice/test19cleanup.service
19:11:15 [    8.687272] testsuite-19.sh[461]: + set +ex
19:11:15 [    8.690223] testsuite-19.sh[462]: Control group /test19cleanup.slice/test19cleanup.service:
19:11:15 [    8.695695] testsuite-19.sh[462]: └─455 sleep infinity
19:11:15 [    8.699776] testsuite-19.sh[463]: FAIL: expected: '1' actual: '0'
```
It seems that `pkill` returns failure? I maybe it's not a race condition that is the problem, but something else.